### PR TITLE
fix: context precedence

### PR DESCRIPTION
**Pull request by MoumitaM**
## Description of the change

> Event-level context will have higher precedence over batch level. default context > event context > batch context

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -253,8 +253,10 @@ pub fn parse_batch(msg:&Batch)-> Ruddermessage{
     for i in &msg.batch {
         match i {
             BatchMessage::Identify(a_) =>{
-                let mut final_context: Value = a_.context.clone().unwrap_or(json!({}));
-                merge(&mut final_context, modified_context.clone() );
+                let mut event_context: Value = a_.context.clone().unwrap_or(json!({}));
+                let mut final_context: Value = batch_context.clone();
+                merge(&mut event_context, default_context.clone());
```
Comment (saikumarrs) [inline src/utils.rs:258]: Are we not allowing default context to be overridden now?

Comment (saikumarrs) [reply src/utils.rs:258]: Sounds good. You've also fixed another bug in the previous implementation.

Comment (saikumarrs) [reply src/utils.rs:258]: However, you're not solving this for the batch-level context. See my other comment.

File: src/utils.rs [hunk 2]
```diff
@@ -253,8 +253,10 @@ pub fn parse_batch(msg:&Batch)-> Ruddermessage{
...
+                merge(&mut final_context, event_context.clone());
```
Comment (saikumarrs) [inline src/utils.rs:259]: This should be the correct implementation.

- Event-level context takes precedence over batch-level context
- Default context takes precedence over any custom context

```suggestion
                let mut event_context: Value = a_.context.clone().unwrap_or(json!({}));
                let mut final_context: Value = batch_context.clone();
                merge(&mut final_context, event_context);
                merge(&mut final_context, default_context.clone());
```

**Review by saikumarrs — CHANGES_REQUESTED** [review]
Don't we have any unit tests to support the library?

File: src/utils.rs (hunk 2 above)
Comment (MoumitaM) [reply src/utils.rs]: Addressed

Comment (MoumitaM) [reply src/utils.rs]: No we don't have unit test case setup

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**